src/coin_test/backtest/simulator.py

```diff
@@ -0,0 +1,232 @@
+        portfolio: Portfolio,
+    ) -> tuple[list[TradeRequest], Portfolio, list[Trade]]:
+        """Process pending orders by adjusting the Portfolio appropriately.
+
+        Args:
+            pending_orders: Uncompleted TradeRequests
+            current_asset_price: Current timestamp's price by AssetPair
+            portfolio: Current Portfolio at given timestamp
+
+        Returns:
+            (remaining pending orders, updated portfolio, executed trades)
+        """
+        executable_orders, pending_orders = Simulator._split_pending_orders(
+            pending_orders, current_asset_price
+        )
+        portfolio, executed_trades = Simulator._execute_orders(
+            portfolio, executable_orders, current_asset_price
+        )
+        return pending_orders, portfolio, executed_trades
+
+    @staticmethod
+    def _strategies_to_run(
+        schedule: Iterable[tuple[Strategy, croniter]],
+        time: dt.datetime,
+        simulation_dt: pd.DateOffset,
+    ) -> list[Strategy]:
+        """Determine which strategies are triggered at a current timestep.
+
+        Args:
+            schedule: The list of strategies to run and their croniters
+            time: The current timestamp
+            simulation_dt: The time interval of the simulation
+
+        Returns:
+            list[Strategy]: The strategies to run this timestep
+        """
+        strategies_to_run = []
+        for strat, cron in schedule:
+            if cron.get_current(dt.datetime) < time + simulation_dt:
+                strategies_to_run.append(strat)
+                while cron.get_current(dt.datetime) < time + simulation_dt:
+                    cron.get_next(dt.datetime)
+
+        return strategies_to_run
+
+    def run_strategies(
+        self,
+        schedule: Iterable[tuple[Strategy, croniter]],
+        time: pd.Timestamp,
+        portfolio: Portfolio,
+    ) -> list[TradeRequest]:
+        """Create TradeRequests for a given timestamp.
+
+        Args:
+            schedule: List of strategies indicating their next run time
+            time: Current timestamp used to determine which strategies should run
+            portfolio: Current Portfolio at given timestamp
+
+        Returns:
+            list of TradeRequests to handle
+        """
+        trade_requests = []
+        for strat in self._strategies_to_run(schedule, time, self._simulation_dt):
+            lookback_data = self._composer.get_range(
+                time - strat.lookback, time, strat.asset_pairs
+            )
+            trade_requests.extend(strat(time, portfolio, lookback_data))
+        return trade_requests
+
+    def run(self) -> tuple[list[Portfolio], list[Trade], list[list[TradeRequest]]]:
+        """Run a simulation."""
+        schedule = [
+            (s, croniter(s.schedule, self._start_time)) for s in self._strategies
+        ]
+
+        # hack fix to remove start time from croniter data
+        for _, cron in schedule:
```

Comment:
I think this skips the first timestep if start time and the croniter string line up? Ex: Star time is a monday and cron string is run every Monday, it seems like this would skip the first Monday.

Reply:
This is @intermezzio's change. I also think this is a bug, Changing to condition that it doesn't match the start time